pkg/nuclide/ui/tree/lib/TreeRootComponent.js

```diff
@@ -205,7 +246,14 @@ var TreeRootComponent = React.createClass({
   },
 
   _onClickNodeArrow(event: SyntheticEvent, node: LazyTreeNode): void {
-    this._toggleNodeExpanded(node);
+    // if alt-clicked the arrow, expand child directories
+    if(event.altKey) {
+      // show/hide child nodes when clicked
+      this._toggleChildNodesExpanded(node, !this._isNodeExpanded(node))
```

Comment:
The ordering here will make expanding a large, remote directory spin for a long time before giving feedback in the UI because this waits to expand the clicked directory until all of its children have expanded. Instead, the clicked directory should expand immediately and then its children should be recursively expanded.

```js
_onClickNodeArrow(...) {
  this._toggleNodeExpanded(node);
  if (event.altKey) {
    this._toggleChildNodesExpanded(node, this._isNodeExpanded(node));
  }
}